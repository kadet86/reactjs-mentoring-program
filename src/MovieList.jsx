import React from 'react';
import MovieListItem from './MovieListItem';

export default function MovieList({ movies }) {
    return (
        <section className="movie-list">
            {movies &&
                movies.map(movie => (
                    <MovieListItem movie={movie} key={movie.id} />
                ))}
        </section>
    );
}
