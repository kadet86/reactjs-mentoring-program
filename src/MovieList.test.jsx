import React from 'react';
// import jest from 'jest';
import {shallow} from 'enzyme';
import MovieList from './MovieList';

describe('MovieList component', () => {
    it('has persistent snapshot with movies={[]}', () => {
        const component = shallow(<MovieList movies={[]}/>);
        expect(component).toMatchSnapshot();
    });

    it('has persistent snapshot with movies={[{id: 1}, {id: 2}, {id: 3}]}', () => {
        const component = shallow(<MovieList movies={[{id: 1}, {id: 2}, {id: 3}]}/>);
        expect(component).toMatchSnapshot();
    });

});
