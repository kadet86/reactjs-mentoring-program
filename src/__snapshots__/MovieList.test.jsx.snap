// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component has persistent snapshot with movies={[]} 1`] = `"<section class=\\"movie-list\\"></section>"`;

exports[`MovieList component has persistent snapshot with movies={[{id: 1}, {id: 2}, {id: 3}]} 1`] = `"<section class=\\"movie-list\\"><div class=\\"movie-list-item\\"><a href=\\"/film/1\\"><img/></a><div><h2></h2><span class=\\"movie-list-item__date\\"></span></div><div class=\\"movie-list-item__genres\\"></div></div><div class=\\"movie-list-item\\"><a href=\\"/film/2\\"><img/></a><div><h2></h2><span class=\\"movie-list-item__date\\"></span></div><div class=\\"movie-list-item__genres\\"></div></div><div class=\\"movie-list-item\\"><a href=\\"/film/3\\"><img/></a><div><h2></h2><span class=\\"movie-list-item__date\\"></span></div><div class=\\"movie-list-item__genres\\"></div></div></section>"`;
